compiler/go.vim

```diff
@@ -21,14 +21,15 @@ else
     CompilerSet makeprg=go\ build
 endif
 
-CompilerSet errorformat=
-      \%-G#\ %.%#,
-      \%-G%.%#panic:\ %m,
-      \%Ecan\'t\ load\ package:\ %m,
-      \%A%f:%l:%c:\ %m,
-      \%A%f:%l:\ %m,
-      \%C%*\\s%m,
-      \%-G%.%#
+" Define the patterns that will be recognized by QuickFix when parsing the output of GoRun.
+" More information at http://vimdoc.sourceforge.net/htmldoc/quickfix.html#errorformat
+CompilerSet errorformat =%-G#\ %.%#                   " Ignore lines beginning with '#' ('# command-line-arguments' line sometimes appears?)
+CompilerSet errorformat+=%-G%.%#panic:\ %m            " Ignore lines containing 'panic: message'
+CompilerSet errorformat+=%Ecan\'t\ load\ package:\ %m " Start of multiline error string is 'can\'t load package'
+CompilerSet errorformat+=%A%f:%l:%c:\ %m              " Start of multiline unspecified string is 'filename:linenumber:columnnumber:'
+CompilerSet errorformat+=%A%f:%l:\ %m                 " Start of multiline unspecified string is 'filename:linenumber:'
+CompilerSet errorformat+=%C%*\\s%m                    " Continuation of multiline error message is indented
+CompilerSet errorformat+=%-G%.%#                      " All lines not matching any of the above patterns are ignored
```

Comment:
Thanks for adding the comments, much appreciated :+1: I'm usually spending 10/20 minutes in `errorformat` before jumping and fixing stuff here :)